### DISCUSSION TITLE
PATCH and rfc 6902

### DISCUSSION THREAD

**Discussion by rma-rripken** [USACE/cwms-data-api#373] [Ideas]
I think we should consider implementing PATCH in our Controllers so that they accept a series of operations.
Something similar to rfc 6902 

Or approx what is described here:
https://williamdurand.fr/2014/02/14/please-dont-patch-like-that/

The changes would be applied in the order specified.  
Further, I would think that we could start a database transaction before performing the operations and then rollback the transaction if there were errors.

The Controller impl would transform the PATCH request body into a series of operation objects.  That series of objects would likely then be passed off to another object that would be responsible for mapping a move operation into a dao rename call.

**Comment by MikeNeilson** [comment]
That sounds like a great idea to me. For now let's just keep it in mind going forward. That doesn't seem hard but it is a fair amount of effort. Plus we need to verify we aren't committing too soon in the database; Daniel has had some issues with that previously. It would make the PATCH operation non-atomic which isn't good.
